api/osb/platform_termination_plugin.go

```diff
@@ -0,0 +1,84 @@
+package osb
+
+import (
+	"fmt"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+	"net/http"
+)
+
+const PlatformTerminationPluginName = "PlatformTerminationPlugin"
+
+type platformTerminationPlugin struct {
+	repository storage.Repository
+}
+
+// NewPlatformTerminationPlugin creates new plugin that checks the platform deletion status
+func NewPlatformTerminationPlugin(repository storage.Repository) *platformTerminationPlugin {
+	return &platformTerminationPlugin{
+		repository: repository,
+	}
+}
+
+// Name returns the name of the plugin
+func (p *platformTerminationPlugin) Name() string {
+	return PlatformTerminationPluginName
+}
+
+// UpdateService intercepts update service instance requests and check if the instance is in the platform from where the request comes
+func (p *platformTerminationPlugin) UpdateService(req *web.Request, next web.Handler) (*web.Response, error) {
+	return p.validateNotPendingTermination(req, next)
+}
+
+// Bind intercepts bind requests and check if the instance is in the platform from where the request comes
+func (p *platformTerminationPlugin) Bind(req *web.Request, next web.Handler) (*web.Response, error) {
+	return p.validateNotPendingTermination(req, next)
+}
+
+func (p *platformTerminationPlugin) Run(req *web.Request, next web.Handler) (*web.Response, error) {
```

Comment:
why do we have Run here? it is not defines in any of the plugin interfaces

Reply:
It's now removed, the OSB tests didn't showed up any regression.